refactor topological_sort to use Node.get_users()

wenzhe-nrv opened the pull request:
replacing deprecated functions: `Node.get_outputs()`
`get_users()` wraps get_output_size and get_output_inputs and returns a NodeVector

rkimballn1 commented on src/ngraph/graph_util.hpp:
```diff
@@ -96,17 +96,11 @@ namespace ngraph
             independent_nodes.pop_front();
 
-            for (auto& output : independent_node->get_outputs())
+            for (auto user : independent_node->get_users())
```
I call auto abuse here. Use `auto&` or better yet `const shared_ptr<Node>&`.  Topological sort is called frequently and creating temporaries can be expensive. Same for line 176

wenzhe-nrv commented:
@rkimballn1 I also found and fixed a few places in graph_util.cpp that used `auto u: node->get_users()`.
Changed to `for (const auto& u : node->get_users())`

wenzhe-nrv commented:
@rkimballn1 bob, please review again.

wenzhe-nrv commented:
This PR is fully reviewed.  Re-requested @louisfeng review by mistake.
